app/views/tag/_sorting.html.erb

```diff
@@ -14,7 +14,8 @@
         items: 8,
         minLength: 3,
         source: function (query, process) {
-          return $.post('/tag/suggested/' + query, {}, function (data) {
+          var replaced_query = query.replace(' ', /-/g);
```

Comment:
Compare the arguments of the `replace` function here with those of the same function on line 30 below. Their types differ. Here the second argument is a regular expression. Shouldn't the first argument be a regular expression?

Reply:
This is true -- if you'd like to file a follow-up issue to make them identical, I'm fine with that! Either of you -- thanks all!

Reply:
@milaaraujo will decide!

Reply:
Hi @jywarren and @seafr! I'm reviewing the code and the arguments I'm using in that `replace` function on line 17 aren't making sense. Sorry! Thank you for noticing that, @seafr.

By typing `'water '`, the substitution would cause an error and the suggested terms were the terms suggested only for `'water'`, without the space. Once the suggestions tags were there and `rake test` did not accuse any error, I thought it was working properly.

Changing the arguments of the replace function on line 17 (for the same ones on line 30), it seems that the replacement of spaces by hyphens happens correctly, but for some reason the list of suggestions does not appear.

![captura de tela 2018-02-27 as 19 27 04](https://user-images.githubusercontent.com/4355017/36768816-c46940b4-1bf5-11e8-8f40-62e815fd9fed.png)

Do you have any idea why this is happening? @jywarren, can I continue working on this issue here? Or should I open a new issue?

Reply:
@milaaraujo have you tried stripping any trailing spaces before the substitution process. 
There is also the fact that `' '` is no the same thing as `\s` as the latter also includes the *tab* and *newline* characters.

Reply:
@milaaraujo have you tried using: `query.replace(/ /g, '-')`?

Reply:
Yep, @seafr. It's the same thing: when I search for 'test test test' on the terminal I see that `Parameters: {"id" => "test-test-test"}`. That's why I think the problem is not the replace function. Now I'm trying to understand how the suggestion process works...

Reply:
:( That's strange. 
The problem is apparently not with the replace function. But the suggestion dropdown window stopped working after you changed it from its previous form: `query.replace(' ', /-/g)`
Let's see what @jywarren has to say on this.